x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -0,0 +1,294 @@
+};
+
+// TODO CREATE_MESSAGE, UPDATE_MESSAGE, and DELETE_MESSAGE from ent-search
+
+export const apiTokenSort = (apiTokenA: IApiToken, apiTokenB: IApiToken) => {
+  if (!apiTokenA.id) {
+    return -1;
+  }
+  if (!apiTokenB.id) {
+    return 1;
+  }
+  return apiTokenA.id - apiTokenB.id;
+};
+
+export interface ICredentialsLogicActions {
+  addEngineName(engineName: string): string;
+  onApiKeyDelete(tokenName: string): string;
+  onApiTokenCreateSuccess(apiToken: IApiToken): IApiToken;
+  onApiTokenError(formErrors: string[]): string[];
+  onApiTokenUpdateSuccess(apiToken: IApiToken): IApiToken;
+  removeEngineName(engineName: string): string;
+  setAccessAllEngines(accessAll: boolean): boolean;
+  setCredentialsData(meta: IMeta, apiTokens: IApiToken[]): { meta: IMeta; apiTokens: IApiToken[] };
+  setCredentialsDetails(details: ICredentialsDetails): ICredentialsDetails;
+  setNameInputBlurred(isBlurred: boolean): boolean;
+  setTokenReadWrite({
+    name,
+    checked,
+  }: {
+    name: string;
+    checked: boolean;
+  }): {
+    name: string;
+    checked: boolean;
+  };
+
+  setTokenName(name: string): string;
+  setTokenType(tokenType: string): string;
+  toggleCredentialsForm(apiToken?: IApiToken): IApiToken;
+  hideCredentialsForm(): { value: boolean };
+  resetCredentials(): { value: boolean };
+  initializeCredentialsData(): { value: boolean };
+  fetchCredentials(page?: number): number;
+  fetchDetails(): { value: boolean };
+  deleteApiKey(tokenName: string): string;
+  onApiTokenChange(): { value: boolean };
+  onEngineSelect(engineName: string): string;
+}
+
+export interface ICredentialsLogicValues {
+  activeApiToken: IApiToken;
+  activeApiTokenIsExisting: boolean;
+  activeApiTokenRawName: string;
+  apiTokens: IApiToken[];
+  apiUrl: string;
+  dataLoading: boolean;
+  engines: IEngine[];
+  // TODO flashMessages from ent-search
+  formErrors: string[];
+  isCredentialsDataComplete: boolean;
+  isCredentialsDetailsComplete: boolean;
+  fullEngineAccessChecked: boolean;
+  keyHelpText: string;
+  meta: IMeta;
+  nameInputBlurred: boolean;
+  showCredentialsForm: boolean;
+}
+
+export const CredentialsLogic = kea<
+  MakeLogicType<ICredentialsLogicValues, ICredentialsLogicActions>
+>({
+  path: ['enterprise_search', 'app_search', 'credentials'],
+  actions: () => ({
+    addEngineName: (engineName) => engineName,
+    onApiKeyDelete: (tokenName) => tokenName,
+    onApiTokenCreateSuccess: (apiToken) => apiToken,
+    onApiTokenError: (formErrors) => formErrors,
+    onApiTokenUpdateSuccess: (apiToken) => apiToken,
+    removeEngineName: (engineName) => engineName,
+    setAccessAllEngines: (accessAll) => accessAll,
+    setCredentialsData: (meta, apiTokens) => ({ meta, apiTokens }),
+    setCredentialsDetails: (details) => details,
+    setNameInputBlurred: (nameInputBlurred) => nameInputBlurred,
+    setTokenReadWrite: ({ name, checked }) => ({
+      name,
+      checked,
+    }),
+    // TODO setFlashMessages from ent-search
```

Comment:
You can delete this line totally. Logic files should generally no longer be in charge of their own flash message states, they should hook into our shared `FlashMessagesLogic` state and update that instead.

Reply:
b6d0fb572072f189c0dd22a607a051fc7ec4b546